src/components/FeaturedData/index.js

```diff
@@ -0,0 +1,248 @@
+  {
+    ssr: false
+  }
+);
+
+const useStyles = makeStyles(({ breakpoints, palette, typography }) => ({
+  root: {
+    flexGrow: 1,
+    backgroundColor: palette.primary.light
+  },
+  layout: {
+    padding: '60px 0',
+    margin: '0 auto',
+    [breakpoints.up('sm')]: {
+      maxWidth: '66.59rem' // .75 of lg
+    },
+    [breakpoints.up('md')]: {
+      maxWidth: '81.3571429rem'
+    }
+  },
+  title: {
+    margin: '1.375rem 0',
+    [breakpoints.up('md')]: {
+      width: '51.125rem'
+    }
+  },
+  chartTitle: {
+    fontFamily: typography.fontFamily,
+    color: '#525e31',
+    fontWeight: 'bold',
+    fontSize: '0.929rem !important'
+  },
+  dullOrange: {
+    color: '#de9f3a'
+  },
+  sourceLink: {
+    fontSize: typography.caption.fontSize
+  },
+  descriptionWrapper: {
+    backgroundColor: '#f1f1ed',
+    padding: '3.42rem 0 0 !important',
+    marginTop: '0 !important',
+    '& div:first-child': {
+      display: 'none'
+    }
+  },
+  description: {
+    fontWeight: '500',
+    fontSize: '0.857rem',
+    lineHeight: '1.92',
+    marginLeft: '0 !important',
+    width: '70%'
+  },
+  chartContainer: {
+    backgroundColor: 'white'
+  },
+  hideHighlightGrid: {
+    display: 'none'
+  }
+}));
+
+function FeaturedData({ selectedCountry }) {
+  const classes = useStyles();
+  const theme = useTheme();
+  const greyChartTheme = {
+    ...theme,
+    chart: {
+      ...theme.chart,
+      colorScale: COLOR_SCALE,
+      pie: {
+        ...theme.chart.pie,
+        colorScale: COLOR_SCALE
+      },
+      area: {
+        ...theme.chart.area,
+        colorScale: COLOR_SCALE
+      },
+      group: {
+        ...theme.chart.group,
+        colorScale: COLOR_SCALE
+      },
+      bar: {
+        ...theme.chart.bar,
+        style: {
+          ...theme.chart.bar.style,
+          data: {
+            ...theme.chart.bar.style.data,
+            fill: COLOR_SCALE[0]
+          }
+        }
+      }
+    }
+  };
+
+  const sectionedCharts = useChartDefinitions();
+  const countryFeaturedCharts = useMemo(
+    () =>
+      featuredCharts[selectedCountry.slug] &&
+      featuredCharts[selectedCountry.slug].length !== 0
+        ? sectionedCharts
+            .map(x => {
+              const charts = x.charts.map(c => {
+                return { ...c, sectionId: x.id };
+              });
+              return charts;
+            })
+            .reduce((a, b) => a.concat(b))
+            .filter(c => {
+              const featuredChartIds = featuredCharts[selectedCountry.slug].map(
+                f => f.id
+              );
+              return featuredChartIds.includes(c.id);
+            })
+            .map(chart => {
+              return {
+                ...chart,
+                description: featuredCharts[selectedCountry.slug].find(
+                  f => f.id === chart.id
+                ).description
+              };
+            })
+        : [],
+    [selectedCountry.slug, sectionedCharts]
+  );
```

Comment:
I think mapping from feature charts can reduce and make the code much more understandable.

```javascript
  const sectionedCharts = useChartDefinitions();
  const countryFeaturedCharts = useMemo(() => {
    const countryFeaturedChartsByIds =
      featuredCharts[selectedCountry.slug] || [];
    const charts = countryFeaturedChartsByIds
      ? sectionedCharts.map(x => x.charts).reduce((a, b) => a.concat(b))
      : [];
    return countryFeaturedChartsByIds.map(f => {
      const chart = charts.find(c => c.id === f.id);
      return { ...chart, ...f };
    });
  }, [selectedCountry.slug, sectionedCharts]);
```

Reply:
@kilemensi  :100: 
But on this `const charts = countryFeaturedChartsByIds
      ? sectionedCharts.map(x => x.charts).reduce((a, b) => a.concat(b))
      : [];` I still have to do an inner map on x.charts to append sectionId to x.charts.

```
const charts = countryFeaturedChartsByIds
      ? sectionedCharts.map(x => { 
        const chart = x.charts.map(z => {...z, sectionId: x.id })
      }).reduce((a, b) => a.concat(b))
      : [];
```